samples/gbmc_socket.py

```diff
@@ -0,0 +1,326 @@
+
+# system parameters
+steps = 50000
+seed = 42
+init_box_l = 7.55
+global_volume = 2.0 * init_box_l**3
+particle_radius = 0.5
+
+global_num_particles = 256
+
+dV_max = 0.2
+kT = 1.25
+
+# LJ-parameters
+lj_epsilon = 1.0
+lj_sigma = 2.0 * particle_radius
+lj_cutoff = 3.0
+lj_shift = 0.0
+
+# Monte-Carlo parameters
+move_chance = 0.5
+exchange_chance = 0.45
+volume_chance = 1.0 - move_chance - exchange_chance
+
+# socket paramters
+HOST = 'localhost'
+PORT = 31415
+number_of_clients = 2
+
+# Message identifiers
+MSG_START = 0
+MSG_END = 1 
+MSG_MOVE_PART = 2
+MSG_MOVE_PART_REVERT = 21
+MSG_CHANGE_VOLUME = 3
+MSG_EXCHANGE_PART_ADD = 4
+MSG_EXCHANGE_PART_ADD_REVERT = 41
+MSG_EXCHANGE_PART_REMOVE = 5
+MSG_EXCHANGE_PART_REMOVE_REVERT = 51
+MSG_ENEGRY = 6
+
+
+class box:
+    '''Box class, which contains the data of one system and coltroles the connection to the respective client.'''
+    box_l = init_box_l
+    box_l_old = init_box_l
+    n_particles = int(global_num_particles / 2)
+    energy = 0.0
+    energy_corrected = 0.0
+    energy_old = 1.0e100
+    conn = 0
+    adr = 0
+
+    def recv_energy(self):
+        '''Recieved the energy data from the client.'''
+        msg = self.recv_data()
+        if(msg[0] == MSG_ENEGRY):
+            self.energy = msg[1]
+            return 0
+        else:
+            print("ERROR during energy recv")
+            return 1
+
+    def recv_data(self):
+        '''Recieved the data send from the client.'''
+        # The first 4 bytes encode the length of the messages recieved.
+        buf = b''
+        while len(buf) < 4:
+            buf += self.conn.recv(4 - len(buf))
+        length = struct.unpack('!I', buf)[0]
+        d = self.conn.recv(length)
+        msg = pickle.loads(d)
+        return(msg)
+
+    def send_data(self, data):
+        '''Send the data packet to the client.'''
+        # The first 4 bytes encode the length of the messages send.
+        length = struct.pack('>I', len(data))
+        packet = length + data
+        self.conn.send(packet)
+
+
+def calc_tail_correction(box):
+    '''
+    Calculates the tail correction to the energies of the box.
+    '''
+    return box.n_particles**2 * 8.0 / 3.0 * np.pi / box.box_l**3 * \
+        (1.0 / 3.0 * np.power(lj_cutoff, -9) - np.power(lj_cutoff, -3))
+
+
+def calc_shift_correction(box):
+    '''
+    Calculates the shift correction to the energies of the box.
+    '''
+    return -1.0 * box.n_particles**2 * 8.0 / 3.0 * np.pi / \
+        box.box_l**3 * np.power(lj_cutoff, 3) * lj_shift
+
+
+# init socket
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+s.bind((HOST, PORT))
+s.listen(number_of_clients)
+
+boxes = []
+p = []
+
+random.seed(seed)
+
+# start clients and connections
+for i in range(number_of_clients):
+    boxes.append(box())
+    p.append(subprocess.Popen(
+        "../pypresso ./gbmc_client.py -n {} -lj {} {} {} {} -s {} -bl {}".format(boxes[i].n_particles,
```

Comment:
please wrap the long list of arguments into compact lines (pep8 is not clever enough)

Reply:
even nicer would be a format string:
```python
f"../pypresso ./gbmc_client.py -n {boxes[i].n_particles} -lj {lj_epsilon} {lj_sigma} {lj_cutoff} {lj_shift} \
    -s {random.randint(0, np.iinfo(np.int32).max)} -bl {boxes[i].box_l}"

Reply:
... but it was first introduced in python 3.6

Reply:
Can we come up with something else than hard coding the path of the pypresso script? Maybe we can provide it as command line argument to the script.